students/john_navitsky/session06/mailroom.py

```diff
@@ -168,13 +182,52 @@ def print_thank_you(donor_id,hint="wonderful",dest=sys.stdout):
     print_lines(2,dest)
     print("-"*80,file=dest)
 
-    print(letter.format(hint,**donors[donor_id]),file=dest)
+    print(letter.format(hint,**current_donor),file=dest)
 
     print("-"*80,file=dest)
     print_lines(2,dest)
 
 
-def thank_you_entry():
+def match_donor(current_donor, donors):
+        """ See if record matches existing record, if so, return full record. """
+        for donor_id in donors:
+            existing_donor=donors[donor_id]
+            if existing_donor["full_name"].lower() == current_donor["full_name"].lower():
+                # if the owner exists, pull in the complete donor record
+                current_donor=existing_donor
+        return current_donor
+
+
+def update_donor(current_donor, donors, new_donation):
```

Comment:
the logic here is a bit unclean. Probably better to do somethign like:

check if the dor is already there.

if it is, get that donor record 

if it isn't, create a new complete donor record and pass that along.

you now have a donor record to work with, and the rest of the code doesn't need to know if it's new or not.

Reply:
Yah, it would probably be better to separate the donor record creation from the record update.